src/ode_solver/runge_kutta_methods/rk_tableau_base.h

```diff
@@ -0,0 +1,40 @@
+#ifndef __RK_TABLEAU_BASE__
+#define __RK_TABLEAU_BASE__
+
+#include "dg/dg.h"
+#include <deal.II/base/conditional_ostream.h>
+
+namespace PHiLiP {
+namespace ODE {
+
+/// Base class for storing the RK method 
+#if PHILIP_DIM==1
+template <int dim, typename real, typename MeshType = dealii::Triangulation<dim>>
+#else
+template <int dim, typename real, typename MeshType = dealii::parallel::distributed::Triangulation<dim>>
+#endif
+class RKTableauBase
+{
+public:
+    /// Default constructor that will set the constants.
+    RKTableauBase(); 
+
+    /// Destructor
+    ~RKTableauBase() {};
+
+    /// Returns Butcher tableau "a" coefficient at position [i][j]
+    virtual double a(int i, int j);
```

Comment:
Unless there is a reason I am unaware of, if you want to keep the code the way it is now, you could make these functions a and b pure virtual in which case you wouldn't need a constructor in rk_tableau_base (similar to src/reduced_order/pod_basis_base.h) and you also wouldn't need a cpp file. This would be an interface or an abstract class. If you do this, you need to define functions a and b in all classes that implement this interface, as you've currently done. This would make the most sense if the functions a and b were defined differently in every class that implements rk_tableau_base. 
However, in this case, the functions a and b are identical in each subclass and the logic is contained only in the constructor. Therefore, it probably makes more sense to factor out the functions a and b and bring them in rk_tableau_base. Then your subclasses would only consist of their constructor.

Reply:
I think your proposal for restructuring the RK tableau classes (i.e., `a()` and `b()` in the base class) would help to reduce repeated code. However, if I did this, there would be risk of segmentation fault - if a user somehow chose `RKTableauBase` as their RK method, there will be a segmentation fault if code tries to access locations in a table that haven't been initialized. Of course, if a user uses the factory they should never be able to choose `RKTableauBase`!
Can you think of a good way to make this safe?

Reply:
Making the constructor of `RKTableauBase` protected should prevent anyone from initializing a `RKTableauBase`, only the subclasses could be initialized.

Reply:
Okay, I will try that -- thanks!

Reply:
```suggestion
    double a(const int i, const int j) const;
```

Reply:
non-virtual since we're not overriding in the derived classes, and const everywhere since the arguments and class variables should not get modified. Also, consider renaming this to `get_a()` since its a basically a "getter". 

Thinking about it more, it may make more sense to have a pure virtual function `virtual set_tableau_a() = 0` (same for b), enforcing that each derived class only "sets" the tableau values. These would be private. If you template this class with `n_rk_stages` and `renit` the tables in the constructor of the base class, followed by `set_tableau_a()` and `set_tableau_b()` in the constructor of the base class, then all the derived classes could have empty constructors and only provide a definition for the `set_tableau` functions. Then use `constexpr` with `n_rk_stages` in the `create_RKTableau` factory

Reply:
I have changed the structure of `RKTableau` and its derived classes to use the setter/getter functionality. I decided against templating n_rk_stages because there shouldn't be a significant performance difference, and it's a bit tidier to pass it as an argument to the constructor rather than as a template.